File: 02. C# Part2/01. Arrays-Homework/04. MaximalSequence/MaximalSequence.cs
using System;
using System.Linq;

    class MaximalSequence
    {
        static void Main()
        {
            //Write a program that finds the maximal sequence of equal elements in an array.

            string input = Console.ReadLine();
            string[] array = input.Split(new char[] { ',', ' ' }, StringSplitOptions.RemoveEmptyEntries);
            int[] nums = new int[array.Length];
            for (int i = 0; i < nums.Length; i++)
            {
                nums[i] = int.Parse(array[i]);
            }
            int[] numbers = input
                .Split(new char[] { ',', ' ' }, StringSplitOptions.RemoveEmptyEntries)
                .Select(n => int.Parse(n)).ToArray();

            int currentNum = nums[0];
            int maxNum = nums[0];
            int currentCount = 0;
            int maxCount = 0;
            for (int i = 0; i < nums.Length;)
            {
                currentNum = nums[i];
                currentCount = 0;
                for (; i < nums.Length; i++)
                {
                    if (currentNum != nums[i])
                    {
                        break;
                    }
                    currentCount++;
                }
                if (maxCount < currentCount)
                {
                    maxCount = currentCount;
                    maxNum = currentNum;
                }
            }
            for (int i = 0; i < maxCount; i++)
            {
                Console.Write(maxNum);
                if (i != maxCount - 1)
                {
                    Console.Write(", ");
                }
            }
            Console.WriteLine();
        }
    }

File: 06. JavaScript Basics/05. Arrays-Homework/05. Arrays-Homework/03. MaximalSequence/MaximalSequence.js
//Write a script that finds the maximal sequence of equal elements in an array.

function sequence(arr) {
    var localCounter = 1;
    var globalCounter = 1;
    var number = 0;
    for (var i = 0; i < arr.length - 1; i++) {
        if (arr[i] === arr[i + 1]) {
            localCounter++;
            if (i === arr.length - 2) {
                globalCounter = localCounter;
                number = i - localCounter + 2;
            }
        }
        else {
            if (localCounter > globalCounter) {
                globalCounter = localCounter;
                number = i - localCounter + 1;
            }

            localCounter = 1;
        }
    }
    var result = arr.slice(number, number + globalCounter);
    jsConsole.writeLine("From the array: [" + arr + "]");
    jsConsole.writeLine("The maximal sequence is: [" + result + "]");
    

}

sequence([2, 1, 1, 2, 3, 3, 2, 2, 2, 1]);
sequence([3, 3, 3, 23, 54, 54, 54, 54]);
sequence([15, 12, 32, 22, 22, 4, 4, 4]);